File: trunk/CreaObjetos/CreaObjetos/Formas/frmCargaXML.vb
Imports System.Xml
Public Class frmCargaXML
    Dim claces As New List(Of Clase)
    Private Sub CargarProyectoToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CargarProyectoToolStripMenuItem.Click
        Dim archivo As String
        Dim ofd As New OpenFileDialog
        Dim m_xmld As XmlDocument
        Dim m_nodelist As XmlNodeList
        Dim m_node As XmlNode
        ofd.ShowDialog()
        archivo = ofd.FileName()

        'Create the XML Reader
        'Try
        m_xmld = New XmlDocument()
        'Disable whitespace so that you don't have to read over whitespaces
        m_xmld.Load(archivo)
        m_nodelist = m_xmld.SelectNodes("/proyecto/clase")

        'Loop through the nodes
        For Each m_node In m_nodelist
            'Get the Gender Attribute Value
            Dim cls As New Clase
            cls.NombreTabla = m_node.Attributes.GetNamedItem("plural").Value
            cls.Neim = m_node.Attributes.GetNamedItem("nombre").Value
            Dim atr_nodeList As XmlNodeList
            Dim atr_node As XmlNode
            atr_nodeList = m_node.SelectNodes("atributo")

            For Each atr_node In atr_nodeList
                Dim propie As New Propiedad
                propie.Nombre = atr_node.InnerText
                propie.Tipo = atr_node.Attributes.GetNamedItem("tipo").Value
                If Not atr_node.Attributes.GetNamedItem("Nulo") Is Nothing Then
                    propie.isNull = True
                Else
                    propie.isNull = False
                End If

                cls.Campos.Add(propie)

            Next
            Me.lsbClases.Items.Add(cls.Neim)
            claces.Add(cls)
        Next
        'Catch errorVariable As Exception
        '    'Error trapping
        '    MessageBox.Show(errorVariable.ToString)
        'End Try




    End Sub

    Private Sub ListBox1_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lsbClases.SelectedIndexChanged
        If claces.Count > 0 And Me.lsbClases.SelectedIndex > -1 Then
            Me.lsbAtributos.DataSource = claces(Me.lsbClases.SelectedIndex).Campos
            Me.lsbAtributos.ValueMember = "nombre"
            Me.lsbAtributos.DisplayMember = "nombre"
        End If
    End Sub

    Private Sub lsbAtributos_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lsbAtributos.SelectedIndexChanged
        If Not Me.lsbAtributos.SelectedItem Is Nothing Then
            Dim prop As Propiedad = CType(Me.lsbAtributos.SelectedItem, Propiedad)
            Me.txtTipo.Text = prop.Tipo
            Me.chbNulo.Checked = prop.isNull
        End If
    End Sub
End Class

File: trunk/CreaObjetos/CreaObjetos.ConvertedToC#/Formas/frmCargaXML.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Xml;
namespace WindowsApplication1
{
	/// <summary>
	/// 
	/// </summary>
	public partial class FrmCargaXml
	{
		List<Clase> claces = new List<Clase>();
		private void CargarProyectoToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
		{
			string archivo = null;
			OpenFileDialog ofd = new OpenFileDialog();
			XmlDocument m_xmld = null;
			XmlNodeList m_nodelist = null;
			XmlNode m_node = null;
			ofd.ShowDialog();
			archivo = ofd.FileName;

			//Create the XML Reader
			//Try
			m_xmld = new XmlDocument();
			//Disable whitespace so that you don't have to read over whitespaces
			m_xmld.Load(archivo);
			m_nodelist = m_xmld.SelectNodes("/proyecto/clase");

			//Loop through the nodes
			foreach (XmlNode m_node_loopVariable in m_nodelist) {
				m_node = m_node_loopVariable;
				//Get the Gender Attribute Value
				Clase cls = new Clase();
				cls.NombreTabla = m_node.Attributes.GetNamedItem("plural").Value;
				cls.Neim = m_node.Attributes.GetNamedItem("nombre").Value;
				XmlNodeList atr_nodeList = null;
				XmlNode atr_node = null;
				atr_nodeList = m_node.SelectNodes("atributo");

				foreach (XmlNode atr_node_loopVariable in atr_nodeList) {
					atr_node = atr_node_loopVariable;
					Propiedad propie = new Propiedad();
					propie.Nombre = atr_node.InnerText;
					propie.Tipo = atr_node.Attributes.GetNamedItem("tipo").Value;
					if ((atr_node.Attributes.GetNamedItem("Nulo") != null)) {
						propie.isNull = true;
					} else {
						propie.isNull = false;
					}

					cls.Campos.Add(propie);

				}
				this.lsbClases.Items.Add(cls.Neim);
				claces.Add(cls);
			}
			//Catch errorVariable As Exception
			//    'Error trapping
			//    MessageBox.Show(errorVariable.ToString)
			//End Try




		}

		private void ListBox1_SelectedIndexChanged(System.Object sender, System.EventArgs e)
		{
			if (claces.Count > 0 & this.lsbClases.SelectedIndex > -1) {
				this.lsbAtributos.DataSource = claces[this.lsbClases.SelectedIndex].Campos;
				this.lsbAtributos.ValueMember = "nombre";
				this.lsbAtributos.DisplayMember = "nombre";
			}
		}

		private void lsbAtributos_SelectedIndexChanged(System.Object sender, System.EventArgs e)
		{
			if ((this.lsbAtributos.SelectedItem != null)) {
				Propiedad prop = (Propiedad)this.lsbAtributos.SelectedItem;
				this.txtTipo.Text = prop.Tipo;
				this.chbNulo.Checked = prop.isNull;
			}
		}
		public FrmCargaXml()
		{
			InitializeComponent();
		}
	}
}
